common/persistence/persistenceMetricClients.go

```diff
@@ -1338,7 +1338,7 @@ func (p *metricEmitter) recordRequestMetrics(operation string, caller string, la
 }
 
 func updateErrorMetric(handler metrics.Handler, logger log.Logger, operation string, err error) {
-	if err != nil {
+	if err != nil && handler != metrics.NoopMetricsHandler {
```

Comment:
why do we need this check? the logic below could also emit logs.

Reply:
I added it to reduce the noisy logs in integration tests, I must have forgotten to remove it.